CMakeLists.txt

```diff
@@ -1,284 +1,41 @@
-if (${LLVM_PACKAGE_VERSION} VERSION_LESS 11)
-  message(FATAL_ERROR "LLVM 11 or newer is required")
-endif()
+include(KLLVMCompilerFlags)
 
 include(SetTriple)
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-
-find_package(ZLIB REQUIRED)
-find_package(FLEX REQUIRED)
-find_package(GMP REQUIRED)
-find_package(Boost REQUIRED COMPONENTS unit_test_framework)
-
-include_directories(${LLVM_INCLUDE_DIRS})
-include_directories (${PROJECT_SOURCE_DIR}/include)
-
-macro(kllvm_add_tool name)
-  add_executable(${name} ${ARGN})
-
-  llvm_config(${name}
-    USE_SHARED true
-    core irreader
-  )
-
-  if(APPLE)
-    if(NOT USE_NIX)
-      target_link_libraries(${name} PUBLIC "-ljemalloc" "-Wl,-rpath ${BREW_PREFIX}/lib" "-ldl")
-    else()
-      target_link_libraries(${name} PUBLIC "-ljemalloc" "-ldl")
-    endif() # NOT USE_NIX
-  else()
-    if(NOT USE_NIX)
-      target_link_libraries(${name} PUBLIC "-ljemalloc" "-Wl,-rpath /usr/local/lib" "-ldl")
-    else()
-      target_link_libraries(${name} PUBLIC "-ljemalloc" "-ldl")
-    endif() # NOT USE_NIX
-  endif() # APPLE
-endmacro(kllvm_add_tool)
-
-configure_file(runtime/main/main.ll ${CMAKE_CURRENT_BINARY_DIR} @ONLY)
-install(
-  FILES ${CMAKE_CURRENT_BINARY_DIR}/main.ll runtime/main/search.cpp
-  DESTINATION lib/kllvm/llvm/main
-)
-
-find_program(OPT NAMES opt-14 opt-13 opt-12 opt-11 opt-10 opt)
-find_program(LLC NAMES llc-14 llc-13 llc-12 llc-11 llc-10 llc)
-if(${OPT} STREQUAL "OPT-NOTFOUND")
-  message(FATAL_ERROR "Could not find an opt binary. Is llvm installed on your PATH?")
-endif()
-if(${LLC} STREQUAL "OPT-NOTFOUND")
-  message(FATAL_ERROR "Could not find an llvm binary. Is llvm installed on your PATH?")
-endif()
-
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
-configure_file(bin/llvm-kompile bin @ONLY)
-configure_file(bin/llvm-krun bin @ONLY)
-configure_file(bin/llvm-kompile-testing bin @ONLY)
-configure_file(bin/llvm-kompile-clang bin @ONLY)
-
-install(
-  PROGRAMS
-    ${CMAKE_CURRENT_BINARY_DIR}/bin/llvm-kompile
-    ${CMAKE_CURRENT_BINARY_DIR}/bin/llvm-kompile-clang
-    ${CMAKE_CURRENT_BINARY_DIR}/bin/llvm-krun
-  DESTINATION bin
-)
-
-install(
-  DIRECTORY include/kllvm
-  DESTINATION include
-)
-
-install(
-  DIRECTORY include/runtime
-  DESTINATION include/kllvm
-)
-install(
-  DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/fmt
-  DESTINATION include/fmt
-)
-install(
-  DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/immer
-  DESTINATION include/kllvm
-)
-install(
-  DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/rapidjson
-  DESTINATION include/kllvm
-)
-
-install(
-  DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/config
-  DESTINATION include/kllvm
-)
-
-file(COPY ${PROJECT_SOURCE_DIR}/deps/immer/immer DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/include)
-file(COPY ${PROJECT_SOURCE_DIR}/deps/fmt/include/fmt DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/include)
-file(COPY ${PROJECT_SOURCE_DIR}/deps/rapidjson/include/rapidjson DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/include)
```

Comment:
Extracted to `deps/CMakeLists.txt` and refactored to a single copy call with relative paths changed